src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const objArr = [];
+  const newState = { ...state };
+
+  for (const a of actions) {
+    switch (a.type) {
+      case 'addProperties':
+        Object.assign(newState, a.extraData);
+        objArr.push({ ...newState });
+        break;
+
+      case 'removeProperties':
+        for (const k of a.keysToRemove) {
+          delete newState[k];
+        }
+        objArr.push({ ...newState });
+        break;
+
+      case 'clear':
+        for (const key in newState) {
+          delete newState[key];
+        }
+        objArr.push({});
```

Comment:
[DONT REPEAT YOURSELF] - Again, pushing the state copy into the 'objArr' is repeated and should be done once after the switch statement. Additionally, pushing an empty object `{}` directly is not necessary, as the newState will already be cleared out by the 'clear' case logic.